docs/recipe_development.rst

```diff
@@ -0,0 +1,132 @@
+==============
+Recipe Development
+==============
+
+mBuild encourages users to subclass ``Compound`` to disseminate reproducible scripts used for building their systems.
```

Comment:
I think the tone of this paragraph is not appropriate for documentation. We may encourage users to publish their scripts, but the focus here should be on how the software enables that process.

I'm also not sure if the pressure to package and publish Python code should be a main focus in this scope; I wonder if the selling point should simply be the accessibility from inside of mBuild